nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,232 @@
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the full-width half maximum in
+        millimeters of the spatial smoothing to apply to the signal.
+
+    standardize: boolean, optional
+        If standardize is True, the time-series are centered and normed:
+        their mean is set to 0 and their variance to 1 in the time dimension.
+
+    detrend: boolean, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details.
+
+    low_pass: False or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details.
+
+    high_pass: False or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details.
+
+    t_r: float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details.
+
+    memory: joblib.Memory or str, optional
+        Used to cache the region extraction process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: int, optional
+        Aggressiveness of memory caching. The higher the number, the higher
+        the number of functions that will be cached. Zero means no caching.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+
+    See also
+    ========
+    nilearn.input_data.NiftiMasker
+    """
+    # memory and memory_level are used by CacheMixin.
+
+    def __init__(self, seeds, radius=None, mask_img=None,
+                 smoothing_fwhm=None, standardize=False, detrend=False,
+                 low_pass=None, high_pass=None, t_r=None,
+                 memory=Memory(cachedir=None, verbose=0), memory_level=1,
+                 verbose=0):
+        self.seeds = seeds
+        self.mask_img = mask_img
+        self.radius = radius
+
+        # Parameters for _smooth_array
+        self.smoothing_fwhm = smoothing_fwhm
+
+        # Parameters for clean()
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+
+        # Parameters for joblib
+        self.memory = memory
+        self.memory_level = memory_level
+
+        self.verbose = verbose
+
+    def fit(self, X=None, y=None):
+        """Prepare signal extraction from regions.
+
+        All parameters are unused, they are for scikit-learn compatibility.
+        """
+            # This is not elegant but this is the easiest way to test it.
+        try:
+            for seed in self.seeds:
+                assert(len(seed) == 3)
+        except Exception as e:
+            if self.verbose > 0:
+                print('Seeds not valid, error' + str(e))
+            raise ValueError('Seeds must be a list of triplets of '
```

Comment:
My version is ugly, I admit, but it catches a lot of bad values. Your version implies that `seeds` is iterable for example.